src/Operators/finitedifference.jl

```diff
@@ -1459,6 +1459,139 @@ end
     return Geometry.Contravariant3Vector(vᶠ * a)
 end
 
+"""
+    U = FCTBorisBook(;boundaries)
+    U.(v, x)
+
+Correct the flux using the flux-corrected transport formulation by Boris and Book
+Input arguments:
+- a face-valued vector field `v`
+- a center-valued field `x`
+```math
+U(v,x)[i] = \\begin{cases}
+  v[i] \\left(-2 x[i-\\tfrac{3}{2}] + 10 x[i-\\tfrac{1}{2}] + 4 x[i+\\tfrac{1}{2}] \\right) / 12  \\textrm{, if } v[i] > 0 \\\\
+  v[i] \\left(4 x[i-\\tfrac{1}{2}] + 10 x[i+\\tfrac{1}{2}] -2 x[i+\\tfrac{3}{2}]  \\right) / 12  \\textrm{, if } v[i] < 0
+  \\end{cases}
+```
+This stencil is based on [BorisBook1973](@cite).
+
+Supported boundary conditions are:
+- [`ThirdOrderOneSided(x₀)`](@ref): uses the third-order downwind reconstruction to compute `x` on the left boundary,
+and the third-order upwind reconstruction to compute `x` on the right boundary.
+
+!!! note
+    Similar to the [`Upwind3rdOrderBiasedProductC2F`](@ref) operator, these boundary conditions do not define the value at the actual boundary faces, and so this operator cannot be materialized directly: it needs to be composed with another operator that does not make use of this value, e.g. a [`DivergenceF2C`](@ref) operator, with a [`SetValue`](@ref) boundary.
+"""
+struct FCTBorisBook{BCS} <: AdvectionOperator
+    bcs::BCS
+end
+FCTBorisBook(; kwargs...) = FCTBorisBook(NamedTuple(kwargs))
+
+return_eltype(::FCTBorisBook, V, A) =
+    Geometry.Contravariant3Vector{eltype(eltype(V))}
+
+return_space(
+    ::FCTBorisBook,
+    velocity_space::Spaces.FaceFiniteDifferenceSpace,
+    arg_space::Spaces.CenterFiniteDifferenceSpace,
+) = velocity_space
+return_space(
+    ::FCTBorisBook,
+    velocity_space::Spaces.FaceExtrudedFiniteDifferenceSpace,
+    arg_space::Spaces.CenterExtrudedFiniteDifferenceSpace,
+) = velocity_space
+
+@inline function fct_boris_book(v, a⁻, a⁺, a⁺⁺, a⁺⁺⁺, step)
+    if v != zero(eltype(a⁻))
+        sign(v) ⊠ (RecursiveApply.rmap(
+            max,
+            zero(eltype(a⁻)),
+            RecursiveApply.rmap(
+                min,
+                RecursiveApply.rmap(abs, v),
+                RecursiveApply.rmap(
+                    min,
+                    sign(v) ⊠ (a⁺⁺⁺ - a⁺⁺) ⊠ step,
+                    sign(v) ⊠ (a⁺ - a⁻) ⊠ step,
+                ),
+            ),
+        ))
+    else
+        RecursiveApply.rmap(
+            max,
+            zero(eltype(a⁻)),
+            RecursiveApply.rmap(
+                min,
+                v,
+                RecursiveApply.rmap(min, (a⁺⁺⁺ - a⁺⁺) ⊠ step, (a⁺ - a⁻) ⊠ step),
+            ),
+        )
+    end
+end
+
+stencil_interior_width(::FCTBorisBook, velocity, arg) =
+    ((0, 0), (-half, half + 2))
+
+@inline function stencil_interior(::FCTBorisBook, loc, idx, hidx, velocity, arg)
+    space = axes(arg)
+    a⁻ = getidx(arg, loc, idx - half, hidx)
+    a⁺ = getidx(arg, loc, idx + half, hidx)
+    a⁺⁺ = getidx(arg, loc, idx + half + 1, hidx)
+    a⁺⁺⁺ = getidx(arg, loc, idx + half + 2, hidx)
```

Comment:
When idx is the  first interior cell face from top,  will `idx + half + 2` go out of the range?

Reply:
No, the infrastructure checks whether we are inside the boundary window or not, [here](https://github.com/CliMA/ClimaCore.jl/blob/b133c3ae65e40c993f46c57466563d8cd1f99faa/src/Operators/finitedifference.jl#L2622-L2644). That said, ensuring that we're not running into bounds errors is not a bad idea to verify. @valeriabarra, can you please add the `bounds-check=yes` to this unit test in the buildkite script? It will make the test run a bit slower, but it'll improve our correctness confidence.